### PR TITLE
Refs #33337 - Reflect Puppet isn't enabled by default

### PR DESCRIPTION
**Pull request by ekohl**
Since Foreman 3.1 Puppet hasn't been enabled in the Katello scenario, but it is in vanilla Foreman. It also rephrases the installer managed parts to avoid the implication that Puppet runs as a deamon.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* [x] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [x] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8; orcharhino 6.3 on EL7/8)
* We do not accept PRs for Foreman older than 3.1.

### CODE REVIEW

File: guides/common/modules/proc_configuring-installation.adoc [hunk 1]
```diff
@@ -20,12 +20,14 @@ An internal label that matches the value is also created and cannot be changed a
 If you do not specify a value, an organization called *Default Organization* with the label *Default_Organization* is created.
 You can rename the organization name but not the label.
 
-* By default, all configuration files configured by the installer are managed by Puppet.
-When `{foreman-installer}` runs, it overwrites any manual changes to the Puppet managed files with the initial values.
+* By default, all configuration files configured by the installer are managed.
+When `{foreman-installer}` runs, it overwrites any manual changes managed files with the initial values.
```
Comment (Lennonka) [inline guides/common/modules/proc_configuring-installation.adoc:24]:
```suggestion
When `{foreman-installer}` runs, it overwrites any manual changes to the managed files with the initial values.
```

**Review by Lennonka — CHANGES_REQUESTED** [review]
This is great! Just a tiny detail.

File: guides/common/modules/proc_configuring-installation.adoc (hunk 1 above)
Comment (ekohl) [reply guides/common/modules/proc_configuring-installation.adoc:24]: That's what you get when you submit a PR in between other work.

File: guides/common/modules/proc_configuring-installation.adoc [hunk 2]
```diff
@@ -20,12 +20,14 @@ An internal label that matches the value is also created and cannot be changed a
...
+ifdef::foreman-el,foreman-deb[]
 By default, {ProjectServer} is installed with the Puppet agent running as a service.
 If required, you can disable Puppet agent on {ProjectServer} using the `--puppet-runmode=none` option.
+endif::[]
 
-* If you want to manage DNS files and DHCP files manually, use the `--foreman-proxy-dns-managed=false` and `--foreman-proxy-dhcp-managed=false` options so that Puppet does not manage the files related to the respective services.
+* If you want to manage DNS files and DHCP files manually, use the `--foreman-proxy-dns-managed=false` and `--foreman-proxy-dhcp-managed=false` options so that the installer does not manage the files related to the respective services.
```
Comment (ekohl) [inline guides/common/modules/proc_configuring-installation.adoc:30]: I was debating if this needs to be pulled into the section started on line 23. On the other hand, I believe we should drop it. We don't have any instructions here to set up DHCP and DNS so isn't it confusing to have this as the initial instructions? I wanted to leave this as a follow up.

Comment (ehelms) [reply guides/common/modules/proc_configuring-installation.adoc:30]: I'd agree, this does seem like a "big" part of the deployment to try to consider here without any additional context that would be better off in the sections on dns / dhcp.

Comment (Lennonka) [reply guides/common/modules/proc_configuring-installation.adoc:30]: I agree to move it. Not sure about removing it entirely.

Comment (ekohl) [reply guides/common/modules/proc_configuring-installation.adoc:30]: Let's have that discussion in https://github.com/theforeman/foreman-documentation/pull/2694.

**Review by Lennonka — APPROVED** [review]

**Comment by Lennonka** [comment]
Cherry-picked both commits to:
- 36bb3138e..70e15d3b8  3.9 -> 3.9
- 04ec292c4..789520586  3.8 -> 3.8

Cherry-picked only the first commit to:
- c14e2e422..dd5ed7893  3.7 -> 3.7
- 47db2355a..0a58e7179  3.6 -> 3.6
- e4cc316cd..35925369c  3.5 -> 3.5
- 0e1969fb8..4debe45c9  3.4 -> 3.4
- 0b512dc95..061fc13b1  3.3 -> 3.3
- 0a49a6d27..cde5e9dfe  3.2 -> 3.2
- 05db08b31..40ab2fd1a  3.1 -> 3.1

The second commit causes conflicts on 3.7 and older. FYI @ekohl

**Pull request merged**